### ISSUE TITLE
A question about the initialization of memory M.

### ISSUE THREAD

**Issue by HX-idiot** [zhmiao/OpenLongTailRecognition-OLTR#8]
Hi, brother, it was great to have a look at your paper recently,
what i confused is ,at the beginning of stage 2, you use average representation for each class as memory, how about just using the fc parameter trained in stage_1?(we can also use cosine classifier in stage_1) Maybe it's more conclusive?  Does there some difference between the two ways?
Thanks~

**Comment by zhmiao** [comment]
Hello @HX-idiot, thanks for asking. Actually, there could be many different ways to construct the memory bank. Feature centroids is one of the most straightforward ways because it is easy to implement and can capture the common invariance within each class that represents the whole class pretty well. At the same time, it fits the use of center loss that makes class clusters more compact and pushes the centroids further away from each other. Weight average is another way for sure. We did not test different types of memories in our paper, but should be an interesting thing to do. 

The reason why we did not apply cosine classifier in the first stage was that the main purpose of the first stage was to initialize the memory bank. We believe that applying cosine classifier at the second stage training is enough for the final classification. However, to further improve the performance, applying cosine to the first stage might help as well.
